### DISCUSSION TITLE
How can I map a shortcut to trigger commit(directly into commit msg insert)

### DISCUSSION THREAD

**Discussion by LintaoAmons** [NeogitOrg/neogit#582] [Q&A] [answered]
So right now, I can `Neogit commit` to toggle the `commit_popup` and then `c` to commit current changes.

And `i` to start insert my commit msg.

How can I map a shortcut to open a new tab to just start to type in my msg(already in insert mode).

Maybe better: to have some setting to put some msg prefix before I start type in some new msg. like `[IssueNo][Name] msg`

**Comment by milkias17** [comment, marked as answer]
Yeah you can set that up in your config by adding `disable_insert_on_commit = false` or you can even set `disable_insert_on_commit = "auto"` to set it up so that if the commit message is empty, go to insert mode, if there's already a commit message for example in an ammend commit, start with normal mode. You can read about it in the readme [here](https://github.com/NeogitOrg/neogit#disabling-insert-on-commit).

**Reply by CKolkey** [reply]
I highly recommend "auto" :)

**Reply by LintaoAmons** [reply]
"auto" is nice
Another question is can I omit popup and jump directly to message buffer?

**Reply by CKolkey** [reply]
Not really - the commit action function takes the popup object as an argument, so you'd have to call a non-public API with an argument that you also don't have access to. Maybe you could figure out a way to mock it, but I consider that private API so no guarantees are made that it won't be broken.

**Comment by gegoune** [comment]
Are you asking to omit popup and jump directly to message buffer?
Don't think that's possible at the moment. 

As for auto adding text, maybe snippets are better solution here?

**Reply by LintaoAmons** [reply]
Yes
